### ISSUE TITLE
Typescript support for `chain` is not working

### ISSUE THREAD

**Issue by vomc** [MoralisWeb3/react-moralis#242]
### Checklist

- [x] I am not disclosing a [vulnerability](https://github.com/MoralisWeb3/react-moralis/blob/main/SECURITY.md).
- [x] I have searched through [existing issues](https://github.com/MoralisWeb3/react-moralis/issues?q=is%3Aissue) and the [Moralis Forum](https://forum.moralis.io/).
- [x] I can reproduce the issue with the [latest react-moralis release](https://github.com/MoralisWeb3/react-moralis/releases), [latest SDK release](https://github.com/MoralisWeb3/Moralis-JS-SDK/releases), and my server is updated to the latest version.

### Issue Description

Try to run the `runContractFunction` example as listed in the moralis docs here: https://docs.moralis.io/moralis-dapp/web3-api/native#runcontractfunction

### Actual Outcome

1. The `useMoralisWeb3ApiCall` function does not exist
2. You get a crashed app because the typing of a string for the chain (e.g. `eth`) will not work, i.e. 
```ts
const options = {
    chain: "eth",
...
```
will result in
```
Type 'string' is not assignable to type '"0x61" | "bsc testnet" | "eth" | "0x1" | "ropsten" | "0x3" | "rinkeby" | "0x4" | "goerli" | "0x5" | "kovan" | "0x2a" | "polygon" | "0x89" | "mumbai" | "0x13881" | "bsc" | "0x38" | ... 6 more ... | undefined'.

```

### Expected Outcome

Somehow I thought that for a 'developer-friendly' toolset the examples in the public documentation would work...

Client

- react-moralis version: `1.3.5`
- Moralis SDK version: `1.6.0`

**Comment by ErnoW** [comment]
This is an issue with Typescript. When you define your `options` object, Typescript will infer the `chain` type as a `string`.

You can avoid this by narrowing the type like:

when the options are static:
```js
const options = {
    chain: 'eth',
    block_number_or_hash: ""
  } as const
```
Or otherwise something similar as this:
```
  const chain: Moralis.Chain = 'eth'
  const options = {
    chain,
    block_number_or_hash: ""
  }
```
You basically want to tell Typescript that `chain` is not just any string, but only a valid `Chain` type

**Comment by x5engine** [comment]
yeah man this is a lame library, when you use typescript you should know how to use it properly
